src/main/java/io/vertx/core/net/impl/SSLHelper.java

```diff
@@ -502,10 +507,37 @@ public SslContext getContext(VertxInternal vertx, String serverName, boolean use
   }
 
   // This is called to validate some of the SSL params as that only happens when the context is created
-  public synchronized void validate(VertxInternal vertx) {
+  public synchronized Future<Void> validate(VertxInternal vertx) {
+    if (validated) {
+      if (validationError != null) {
+        return Future.failedFuture(validationError);
+      }
+      return Future.succeededFuture();
+    }
+
+    validated = true;
+
     if (ssl) {
-      getContext(vertx, null);
+      ContextInternal validateContext = vertx.getOrCreateContext();
```

Comment:
the context should be provided as argument of the validate method instead of being access as a static.

Reply:
Done